R/prepare_ae_specific_subgroup.R

```diff
@@ -1,62 +1,45 @@
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
 #
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
 #
-# You should have received a copy of the GNU General Public License
-# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #' Prepare datasets for AE specific analysis
 #'
 #' @inheritParams prepare_ae_specific
-#' @param subgroup_var A character value of subgroup variable name in
-#'   observation data saved in `meta$data_observation`.
-#' @param subgroup_header An integer value of column header for subgroup.
-#' @param display_total test
-#' @param display_subgroup_total test
-#'
-#' @return A list of analysis raw datasets.
-#'
-#' @export
-#'
+#' @param subgroup_var a character value of subgroup variable name in observation data saved in `meta$data_observation`.
+#' @param column_hierarchy a character value for type of subgroup analysis.
+#' @param display_subgroup_total a logical Value to display total column for subgroup analysis.
 #' @examples
-#' meta <- meta_ae_example()
-#'
-#' prepare_ae_specific_subgroup(
-#'   meta,
-#'   population = "apat",
-#'   observation = "wk12",
-#'   parameter = "rel",
-#'   subgroup_var = "RACE"
-#' )
+#' # meta <- meta_ae_dummy()
+#' # prepare_ae_specific_subgroup(meta, "apat", "wk12", "rel", subgroup_var = "RACE")$data
+#' @export
 prepare_ae_specific_subgroup <- function(meta,
                                          population,
                                          observation,
                                          parameter,
                                          subgroup_var,
-                                         subgroup_header = 1,
+                                         column_hierarchy = "trt_within_sub",
                                          components = c("soc", "par"),
-                                         display_total = TRUE,
                                          display_subgroup_total = TRUE) {
-  message("This function is still under development.")
-  return(invisible(NULL))
-
-  if (!subgroup_header %in% c(1, 2)) {
-    stop("`subgroup_header` can only have value 1 or 2.", call. = FALSE)
+  if (!column_hierarchy %in% c("trt_within_sub", "sub_within_trt")) {
```

Comment:
You will want to write

```r
!(column_hierarchy %in% c("trt_within_sub", "sub_within_trt"))
```

to make the code easy to reason about because I would have thought `column_hierarchy` is a logical but it is not.